src/test/regress/sql/alter_table_ao.sql

```diff
@@ -447,3 +447,38 @@ CREATE table ao_insert_empty_row (a integer, b text, c integer) WITH (APPENDONLY
 INSERT INTO ao_insert_empty_row SELECT 1,'a',1 FROM gp_id WHERE dbid=-999;
 -- assert non-empty seg entries
 SELECT * FROM gp_toolkit.__gp_aoseg('ao_insert_empty_row'::regclass);
+
+-- test case: alter AO table add partition, the preference of the storage setting is: the encoding clause > table setting > gp_default_storage_options
+SET gp_add_partition_inherits_table_setting = on;
+
+CREATE TABLE ao_alter_add_part(a int, b int, c int) WITH (appendonly=true, orientation=row, compresstype=zstd, compresslevel=4, blocksize=65536) DISTRIBUTED BY (a)
+    PARTITION BY RANGE (b)
+        SUBPARTITION BY LIST (c) SUBPARTITION TEMPLATE (SUBPARTITION "11" VALUES (1, 2, 3) WITH (appendonly=true, orientation=row, compresstype=zstd, compresslevel=5, blocksize=32768))
```

Comment:
Let's add `SUBPARTITION "22" VALUES (4, 5, 6) WITH (appendonly=true, orientation=row)` to check that subpartitions inherit storage options too

Reply:
added